src/utils/transform.ts

```diff
@@ -197,3 +216,283 @@ function valueToTag(key: string, value: string | number | undefined, type: strin
+        config: { unit: 't/min', displayName: 'Transactions per minute' },
+      };
+
+  const servicesMap: { [refId: number]: XrayService } = {};
+  const edges: Array<{
+    edge: XrayEdge;
+    source: XrayService;
+  }> = [];
+
+  for (const service of services) {
+    const statsSource = service.SummaryStatistics ? service : service.Edges[0];
+    const stats = statsSource.SummaryStatistics;
+
+    idField.values.add(service.ReferenceId);
+    titleField.values.add(service.Name);
+    typeField.values.add(service.Type);
+    mainStatField.values.add(avgResponseTime(stats));
+
+    if (showRequestCounts) {
+      const count = statsSource.ResponseTimeHistogram.reduce((acc, h) => acc + h.Count, 0);
+      secondaryStatField.values.add(count + ' Request' + (count > 1 ? 's' : ''));
+    } else {
+      secondaryStatField.values.add(
+        service.SummaryStatistics
+          ? tracesPerMinute(stats, service.StartTime, service.EndTime)
+          : // For root nodes we compute stats from it's edge
+            tracesPerMinute(stats, service.Edges[0].StartTime, service.Edges[0].EndTime)
+      );
+    }
+    successField.values.add(successPercentage(stats));
+    errorsField.values.add(errorsPercentage(stats));
+    faultsField.values.add(faultsPercentage(stats));
+    throttledField.values.add(throttledPercentage(stats));
+
+    servicesMap[service.ReferenceId] = service;
+    edges.push(...service.Edges.map(e => ({ edge: e, source: service })));
+  }
+
+  for (const edgeData of edges) {
+    const { edge, source } = edgeData;
+    const target = servicesMap[edge.ReferenceId];
+    edgeIdField.values.add(source.ReferenceId + '__' + target.ReferenceId);
+    edgeSourceField.values.add(source.ReferenceId);
+    edgeTargetField.values.add(edge.ReferenceId);
+    edgeSourceNameField.values.add(source.Name);
+    edgeTargetNameField.values.add(target.Name);
+
+    const stats = edge.SummaryStatistics;
+
+    const success = successPercentage(edge.SummaryStatistics);
+    if (success === 1) {
+      edgeMainStatField.values.add(`Success ${(success * 100).toFixed(2)}%`);
+    } else {
+      const firstNonZero = ([
+        [faultsPercentage(stats), 'Faults'],
+        [errorsPercentage(stats), 'Errors'],
+        [throttledPercentage(stats), 'Throttled'],
+      ] as Array<[number, string]>).find(v => v[0] !== 0);
+      if (!firstNonZero) {
+        edgeMainStatField.values.add(`N/A`);
+      } else {
+        edgeMainStatField.values.add(`${firstNonZero[1]} ${(firstNonZero[0] * 100).toFixed(2)}%`);
+      }
+    }
+
+    if (showRequestCounts) {
+      const count = edge.ResponseTimeHistogram.reduce((acc, h) => acc + h.Count, 0);
+      edgeSecondaryStatField.values.add(count + ' Request' + (count > 1 ? 's' : ''));
+    } else {
+      edgeSecondaryStatField.values.add(tracesPerMinute(edge.SummaryStatistics, edge.StartTime, edge.EndTime));
+    }
+  }
+
+  for (let i = 0; i < edgeTargetField.values.length; i++) {}
```

Comment:
I think this is not needed?

Reply:
Yeah does not seem like the most important part of the logic :D